pkgs/misc/uboot/hardkernel-firmware.nix

```diff
@@ -0,0 +1,126 @@
+      } // extraMeta;
+    } // removeAttrs args [ "extraMeta" ]);
+in {
+  inherit buildHardkernelFirmware;
+
+  # https://wiki.odroid.com/odroid-c2/software/building_u-boot
+  # $ nix-build -A pkgsCross.aarch64-multiplatform.firmwareOdroidC2
+  firmwareOdroidC2 = let
+    name = "odroidc2";
+    odroidc2-bl301 = fetchgit {
+      url = "https://github.com/hardkernel/u-boot_firmware.git";
+      rev = "b7b90c1099b057d35ebae886b7846b5d9bfb4143"; # "odroidc2-bl301"
+      sha256 = "0kdb1mg5zd7qyabfpbh98cs07icfzpkywvva13ybf4mf5g50g0n3";
+      deepClone = true;
+    };
+  in buildHardkernelFirmware {
+    defconfig = "odroidc2_config";
+    name = "firmware-odroid-c2";
+    version = "2015.01";
+    src = fetchgit {
+      url = "https://github.com/hardkernel/u-boot.git";
+      rev = "fac4d2da0a1b61dfdeaca0034a45151ff5983fb8"; # "odroidc2-v2015.01"
+      sha256 = "1wslpg7nrbzqns1gglkxgi1ka2rpmpmhigk4p4f4589lkky4jrmd";
+      leaveDotGit = true;
+    };
+
+    prePatch = ''
+      git remote add u-boot_firmware ${odroidc2-bl301}/
+      git fetch u-boot_firmware
+      git cherry-pick --no-commit \
+        5ce504067bb83de03d17173d5585e849df5d5a33^..${odroidc2-bl301.rev}
+
+      substituteInPlace ./arch/arm/cpu/armv8/gxb/firmware/scp_task/Makefile \
+        --replace "CROSS_COMPILE" "CROSS_COMPILE_32"
+    '';
+
+    postBuild = ''
+      patchelf --set-interpreter "$(cat ${buildPackages.stdenv.cc}/nix-support/dynamic-linker)" fip/fip_create
+    '';
+
+    filesToInstall = [
+      "build/scp_task/bl301.bin" "fip/fip_create" "fip/gxb/aml_encrypt_gxb"
+      "fip/gxb/bl30.bin" "fip/gxb/bl31.bin" "fip/gxb/bl2.package"
+      "sd_fuse/sd_fusing.sh" "sd_fuse/bl1.bin.hardkernel"
+    ];
+
+    extraMeta.platforms = [ "aarch64-linux" ];
+  };
+
+  # https://wiki.odroid.com/odroid-c4/software/building_u-boot
+  # $ nix-build -A pkgsCross.aarch64-multiplatform.firmwareOdroidC4
+  firmwareOdroidC4 = buildHardkernelFirmware {
+    name = "firmware-odroid-c4";
+    defconfig = "odroidc4_defconfig";
+    version = "2015.01";
+    src = fetchgit {
+      url = "https://github.com/hardkernel/u-boot.git";
+      rev = "2a4e31e3fde9bced8a150d338aa397a0362df191"; # "odroidg12-v2015.01"
+      sha256 = "0wgw5i1wv6kg0h06glndgfafiqi5p8pkxprnf1ipbhlpyb4mz4zn";
+      leaveDotGit = true;
+    };
+
+    prePatch = ''
+      substituteInPlace ./arch/arm/cpu/armv8/g12a/firmware/scp_task/Makefile \
+        --replace "CROSS_COMPILE" "CROSS_COMPILE_32"
+    '';
+
+    filesToInstall = [
+      "build/board/hardkernel/odroidc4/firmware/acs.bin" "build/scp_task/bl301.bin"
+      "fip/g12a/bl2.bin" "fip/g12a/bl30.bin" "fip/g12a/bl31.img"
+      "fip/g12a/ddr3_1d.fw" "fip/g12a/ddr4_1d.fw" "fip/g12a/ddr4_2d.fw"
+      "fip/g12a/lpddr3_1d.fw" "fip/g12a/lpddr4_1d.fw" "fip/g12a/lpddr4_2d.fw"
+      "fip/g12a/diag_lpddr4.fw" "fip/g12a/piei.fw" "fip/g12a/aml_ddr.fw"
+      "fip/g12a/aml_encrypt_g12a" "sd_fuse/sd_fusing.sh"
+    ];
+
+    extraMeta.platforms = [ "aarch64-linux" ];
```

Comment:
I think `x86_64-linux` makes more sense here because all the precompiled Linux binaries are for x86_64. Then you can use it through `buildPackages`. Also, add a comment explaining this unusual sitation.

Reply:
ditto